### PR TITLE
Move job and tasks logs into a job_id folder

### PR DESCRIPTION
**Pull request by marcocast**

### CODE REVIEW

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/JobLogger.java [hunk 1]
```diff
@@ -90,7 +90,7 @@ private void updateMdcWithTaskLogFilename(JobId id) {
     }
 
     public static String getJobLogFilename(JobId id) {
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/JobLogger.java]: maybe rename this method to getJobLogRelativePath()

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ServerJobAndTaskLogs.java [hunk 2]
```diff
@@ -84,28 +84,26 @@ public static String getJobLog(JobId jobId, Set<TaskId> tasks) {
...
     private static void removeFile(String path) {
         if (logsLocationIsSet()) {
             String logsLocation = getLogsLocation();
-            File f = new File(logsLocation, path);
-            if (f.exists()) {
-                logger.info("Removing file " + f.getAbsolutePath());
-                f.delete();
+            File logFolder = new File(logsLocation, path);
+            try {
+                org.apache.commons.io.FileUtils.deleteDirectory(logFolder);
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ServerJobAndTaskLogs.java]: It is probably better to use deleteQuietly, throwing an exception here would be worse than the actual file deletion problem.

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ServerJobAndTaskLogs.java [hunk 3]
```diff
@@ -84,28 +84,26 @@ public static String getJobLog(JobId jobId, Set<TaskId> tasks) {
...
+    public static void remove(JobId jobId) {
+        removeFolderLog(jobId.value());
     }
 
-    private static void removeLog(String filename) {
-        for (String suffix : new String[] { "", ".1" }) {
-            removeFile(filename + suffix);
-        }
+    private static void removeFolderLog(String folderName) {
+        removeFile(folderName);
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/ServerJobAndTaskLogs.java]: if removeFile is not called somewhere else then it's better to merge the two methods and call it removeLogFolder

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/TaskLogger.java [hunk 4]
```diff
@@ -119,7 +119,7 @@ private void updateMdcWithTaskLogFilename(TaskId id) {
     }
 
     public static String getTaskLogFilename(TaskId id) {
-        return id.toString();
+        return id.getJobId().value() + "/" + id.toString();
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/util/TaskLogger.java]: same remark as above, call this method getTaskLogRelativePath()

**Comment by marcocast** [comment]
retest this please

**Comment by marcocast** [comment]
retest this please

**Comment by marcocast** [comment]
retest this please

**Pull request merged**